src/mapnik_vector_tile.cpp

```diff
@@ -4404,13 +4415,38 @@ v8::Local<v8::Value> VectorTile::_getDataSync(Nan::NAN_METHOD_ARGS_TYPE info)
             }
             if (!compress)
             {
-                return scope.Escape(Nan::CopyBuffer((char*)d->tile_->data(),raw_size).ToLocalChecked());
+                if (release)
+                {
+                    std::unique_ptr<std::string> out = d->tile_->release_buffer();
+                    return scope.Escape(Nan::NewBuffer(&((*out)[0]),
+                                                       out->size(),
+                                                       [](char*, void* hint) {
+                                                           delete reinterpret_cast<std::string*>(hint);
+                                                       },
+                                                       out.release())
```

Comment:
The may cause dereferencing a null pointer. The order of evaluation of function call arguments is unspecified. `out.release()` may be called before `out->size()`.

https://en.cppreference.com/w/cpp/language/eval_order

Reply:
@lightmare thank you so much for noticing this! I will try something else out.